### PR TITLE
Add JBoss EAP clustered VMSS RHEL

### PR DESCRIPTION
**Pull request by Suraj2093**
# PR Checklist

Check these items before submitting a PR... 

[Contribution Guide](https://github.com/Azure/azure-quickstart-templates/blob/master/1-CONTRIBUTION-GUIDE/README.md)

[Best Practice Guide](https://github.com/Azure/azure-quickstart-templates/blob/master/1-CONTRIBUTION-GUIDE/best-practices.md)

- [x] - Please check this box once you've submitted the PR if you've read through the Contribution Guide and best practices checklist.

## Changelog

*  Add template and all files required for JBoss EAP clustered VMSS RHEL

### CODE REVIEW

File: jboss-eap-clustered-vmss-rhel/README.md [hunk 1]
```diff
@@ -0,0 +1,336 @@
...
+## Deployment Time
+
+The deployment takes approximately 10 minutes to complete.
+
+## Validation Steps
+
+Once the deployment is successful, go to the outputs section of the deployment to obtain the **app URL**. You can access the RHEL VMSS instance and the application by following one of the 3 options:
+
+1. Create a Jump VM in a different subnet (new subnet) in the same Virtual Network and access the Load Balancer and RHEL VMSS instance via Jump VM.
+2. Create a Jump VM in a different Virtual Network and access the Load Balancer and RHEL VMSS instance using Virtual Network Peering. (recommended method)
```
Comment (bmoore-msft) [inline jboss-eap-clustered-vmss-rhel/README.md:184]: put the recommended option first

File: jboss-eap-clustered-vmss-rhel/README.md [hunk 2]
```diff
@@ -0,0 +1,336 @@
...
+
+Your Private Offer will be tagged with *Private* as seen below.
+
+![alt text](images/rhel-byos.png)
+
+Please refer to [Using the Azure Custom Script Extension Version 2 with Linux VMs](https://docs.microsoft.com/azure/virtual-machines/extensions/custom-script-linux) for more details on troubleshooting VM custom script extensions.
+
+#### Cleanup
+
+If your deployment fails due to any reason (like the ones mentioned above), make sure you delete the whole resource group so that you are not charged for any of the resources deployed so far before failure occurred. In case of successful deployment, follow the same process of deleting the resource group created by the template and related resources (if created in other resource groups) to optimize Azure cost. Once your resources are deleted make sure you log into your RHSM account and remove the system you registered for the deployment of the template, so that you do not hit the "Maximum Enabled Entitlement Quantity" (if you have set any). You can remove the registered system under the *Systems* section in the Red Hat portal.
```
Comment (bmoore-msft) [inline jboss-eap-clustered-vmss-rhel/README.md:330]: if I have other resources in the resourceGroup, I probably don't want to delete those

**Review by bmoore-msft — CHANGES_REQUESTED** [review]

File: jboss-eap-clustered-vmss-rhel/azuredeploy.json [hunk 3]
```diff
@@ -0,0 +1,579 @@
...
+          }
+        ]
+      }
+    },
+    "planBYOS": {
+      "name": "[if(equals(parameters('eapOnRHELVersion'),'JBoss-EAP7.2-on-RHEL7.7'), 'rhel-lvm77', 'rhel-lvm8')]",
+      "publisher": "redhat",
+      "product": "rhel-byos"
+    },
+    "storageProfilePAYG": {
```
[the remark this answers is not in the document]
Comment (Suraj2093) [reply jboss-eap-clustered-vmss-rhel/azuredeploy.json]: Reduced the duplication and made the samples more readable as you have mentioned above

File: jboss-eap-clustered-vmss-rhel/azuredeploy.json [hunk 4]
```diff
@@ -0,0 +1,579 @@
...
+    "scriptFolder": "scripts",
+    "fileToBeDownloaded": "eap-session-replication.war",
+    "scriptArgs": "[concat('-a ', uri(parameters('_artifactsLocation'), '.'), ' -t \"', parameters('_artifactsLocationSasToken'), '\" -p ', variables('scriptFolder'), ' -f ', variables('fileToBeDownloaded'))]"
+  },
+  "resources": [
+    {
+      "condition": "[equals(parameters('eapOnRHELVersion'),'JBoss-EAP7.2-on-RHEL7.7')]",
+      "type": "Microsoft.Resources/deployments",
+      "apiVersion": "2019-05-01",
+      "name": "pid-92abde63-f331-4e50-96ca-e0e5c2339caa",
```
[the remark this answers is not in the document]
Comment (Suraj2093) [reply jboss-eap-clustered-vmss-rhel/azuredeploy.json]: Removed duplicate resource

File: jboss-eap-clustered-vmss-rhel/azuredeploy.json [hunk 5]
```diff
@@ -0,0 +1,579 @@
...
+            "name": "[variables('subnetName')]",
+            "properties": {
+              "addressPrefix": "[variables('subnetPrefix')]"
+            }
+          }
+        ]
+      }
+    },
+    {
+      "condition": "[equals(parameters('rhelOSLicenseType'),'PAYG')]",
```
[the remark this answers is not in the document]
Comment (Suraj2093) [reply jboss-eap-clustered-vmss-rhel/azuredeploy.json]: This is done

File: jboss-eap-clustered-vmss-rhel/README.md [hunk 6]
```diff
@@ -0,0 +1,336 @@
...
+
+[![Deploy To Azure](https://raw.githubusercontent.com/Azure/azure-quickstart-templates/master/1-CONTRIBUTION-GUIDE/images/deploytoazure.svg?sanitize=true)](https://portal.azure.com/#create/Microsoft.Template/uri/https%3A%2F%2Fraw.githubusercontent.com%2FAzure%2Fazure-quickstart-templates%2Fmaster%2Fjboss-eap-clustered-vmss-rhel%2Fazuredeploy.json)
+[![Deploy To Azure US Gov](https://raw.githubusercontent.com/Azure/azure-quickstart-templates/master/1-CONTRIBUTION-GUIDE/images/deploytoazuregov.svg?sanitize=true)](https://portal.azure.us/#create/Microsoft.Template/uri/https%3A%2F%2Fraw.githubusercontent.com%2FAzure%2Fazure-quickstart-templates%2Fmaster%2Fjboss-eap-clustered-vmss-rhel%2Fazuredeploy.json)
+[![Visualize](https://raw.githubusercontent.com/Azure/azure-quickstart-templates/master/1-CONTRIBUTION-GUIDE/images/visualizebutton.svg?sanitize=true)](http://armviz.io/#/?load=https%3A%2F%2Fraw.githubusercontent.com%2FAzure%2Fazure-quickstart-templates%2Fmaster%2Fjboss-eap-clustered-vmss-rhel%2Fazuredeploy.json)
+
+`Tags: JBoss, Red Hat, EAP 7.2, EAP 7.3, Cluster, Load Balancer, RHEL 7.7, RHEL 8.0, Azure, Azure VMSS, Java EE`
```
[the remark this answers is not in the document]
Comment (Suraj2093) [reply jboss-eap-clustered-vmss-rhel/README.md]: Moved the tags to bottom

File: jboss-eap-clustered-vmss-rhel/README.md [hunk 7]
```diff
@@ -0,0 +1,336 @@
...
+## Subscriptions and Costs
+
+This ARM template is designed with flexible operating system (OS) options:
+
+- Red Hat Enterprise Linux OS as Pay-As-You-Go (PAYG) or Bring-Your-Own-Subscription (BYOS) via Red Hat Gold Image model.
+- Red Hat JBoss EAP is available through BYOS only.
+
+#### Using RHEL OS with PAYG Model
+
+By default this template uses the on-demand Red Hat Enterprise Linux PAYG image from the Azure Gallery. When using this on-demand image, there is an additional hourly RHEL subscription charge for using this image on top of the normal compute, network and storage costs. At the same time, the instance will be registered to your Red Hat subscription, therefore consuming one of your entitlements. This will lead to "double billing". To avoid this, you would need to build your own RHEL image, which is defined in this Red Hat KB article for [uploading RHEL image to Azure](https://access.redhat.com/articles/uploading-rhel-image-to-azure) or use RHEL Gold Image from the Azure Private Gallery offering.
```
[the remark this answers is not in the document]
Comment (Suraj2093) [reply jboss-eap-clustered-vmss-rhel/README.md:69]: If the user does not want to follow the extra steps to deploy BYOS license RHEL VM, they can deploy the PAYG as mentioned in the QS sample.

File: jboss-eap-clustered-vmss-rhel/README.md (hunk 1 above)
Comment (Suraj2093) [reply jboss-eap-clustered-vmss-rhel/README.md]: Changed the order as recomended

File: jboss-eap-clustered-vmss-rhel/README.md (hunk 2 above)
Comment (Suraj2093) [reply jboss-eap-clustered-vmss-rhel/README.md]: Added a note about the same

**Comment by Suraj2093** [comment]
I have made the changes that you have suggested and added comments. Please review and approve the PR ASAP.

**Pull request merged**